pkg/e2etest/e2e_test.go

```diff
@@ -0,0 +1,529 @@
+
+	// Summary consistency check (do not fail yet to show details in the next check)
+	if !reflect.DeepEqual(resReference.Summary, resGoQuery.Summary) {
+		t.Errorf("Mismatch on goQuery summary, want %+v, have %+v", resReference.Summary, resGoQuery.Summary)
+	}
+
+	// Since testify creates very unreadable output when comparing the struct directly we build a stringified
+	// version of the result rows and compare that
+	refRows := make([]string, len(resReference.Rows))
+	for i := 0; i < len(refRows); i++ {
+		refRows[i] = fmt.Sprintf("%s (%s): %s %s",
+			resReference.Rows[i].Labels.Hostname, resReference.Rows[i].Labels.HostID,
+			resReference.Rows[i].Attributes,
+			resReference.Rows[i].Counters)
+	}
+	resRows := make([]string, len(resGoQuery.Rows))
+	for i := 0; i < len(resRows); i++ {
+		resRows[i] = fmt.Sprintf("%s (%s): %s %s",
+			resGoQuery.Rows[i].Labels.Hostname, resGoQuery.Rows[i].Labels.HostID,
+			resGoQuery.Rows[i].Attributes,
+			resGoQuery.Rows[i].Counters)
+	}
+	require.EqualValues(t, refRows, resRows)
+}
+
+func runGoProbe(t *testing.T, sourceInitFn func() (mockIfaces, func(c *capture.Capture) (slimcap.Source, error))) {
+
+	// We quit on encountering SIGUSR2 (instead of the ususal SIGTERM or SIGINT)
+	// to avoid killing the test
+	ctx, stop := signal.NotifyContext(context.Background(), syscall.SIGUSR2)
+	defer stop()
+
+	mockIfaces, initFn := sourceInitFn()
+	captureManager := capture.NewManager(ctx).SetSourceInitFn(initFn)
+
+	ifaceConfigs := make(config.Ifaces)
+	for _, iface := range mockIfaces {
+		ifaceConfigs[iface.name] = defaultCaptureConfig
+	}
+
+	// Wait until goProbe is done processing all packets, then kill it in the
+	// background via the SIGUSR2 signal
+	go mockIfaces.KillGoProbeOnceDone(captureManager)
+
+	captureManager.Update(ifaceConfigs, nil)
+
+	// Start goroutine for writeouts
+	writeoutHandler := writeout.NewHandler(captureManager, encoders.EncoderTypeLZ4).
+		WithPermissions(goDB.DefaultPermissions)
+
+	// Start writeout handler
+	writeoutChan := writeoutHandler.HandleWriteouts()
+
+	// Start regular rotations (minimum one second to avoid timestamp duplicates)
+	writeoutHandler.HandleRotations(ctx, time.Second)
+
+	// Wait for the interrupt signal
+	<-ctx.Done()
+
+	// Finish up
+	writeoutHandler.Close()
+	<-writeoutChan
+	captureManager.CloseAll()
+}
+
+func runGoQuery(t *testing.T, mockIfaces mockIfaces, maxEntries int) results.Result {
+
+	buf := bytes.NewBuffer(nil)
+	old := os.Stdout // keep backup of STDOUT
+	r, wr, err := os.Pipe()
+	require.Nil(t, err)
+	os.Stdout = wr
+
+	// copy the output in a separate goroutine so printing can't block indefinitely
+	copyDone := make(chan struct{})
+	go func() {
+		if _, err := io.Copy(buf, r); err != nil {
+			panic(err)
+		}
+		copyDone <- struct{}{}
+	}()
+
+	command := cmd.GetRootCmd()
+	command.SetArgs([]string{
+		"-i", strings.Join(mockIfaces.Names(), ","),
+		"-e", "json",
+		"-l", time.Now().Add(time.Hour).Format(time.ANSIC),
```

Comment:
This is done similarly for the default of the argument inside root.go. Or is there a reason that you add an hour?

Reply:
Due to the short time periods the "fake" goProbe runs I encountered that it was missing the last timestamp in about half of the cases (because `Now()` at the point of the call cuts _before_ the last timestamp if it's the same (which happens, statistically speaking, about half of the time). +1s would probably we enough here, I just wanted to tell `goQuery`: "all" in a very clear way :stuck_out_tongue_closed_eyes: